modules/mod_sources/mod_source_base.ts

```diff
@@ -1,12 +1,112 @@
+import ElectronLog from 'electron-log';
+import Main from '../../main';
+import path from 'path';
+import fs from "fs"
+import {Install, ModListEntry} from '../mod_list_loader'
+import {FindCollectionNumber} from '../mod_manager'
+
+
 abstract class ModInstallSource {
-    data: any;
+    data: Install[];
     fileType = "UNKNOWN";
 
-    constructor(install_data){
+    constructor(install_data : Install[]){
         this.data = install_data;
     }
     abstract GetLatestVersionNumber() : Promise<number>;
     abstract GetDisplayVersionNumber(): Promise<string>;
-    abstract GetFileURL() : Promise<string>;
+    abstract GetFileURL(asset_index? : number) : Promise<string>;
+
+    PostInstall(collection_version?: string) {
+        //Try to execute mod specific operations, like moving tf/cfg/class.cfg and tf/cfg/autoexec.cfg back to /tf/cfg/user/class.cfg
+        //and /tf/cfg/user/autoexec.cfg respectively for mastercomfig
+
+        let setup_func: string = "";
+        if (!(typeof (this.data[FindCollectionNumber(this.data, collection_version)].setupfunc) == "undefined")) {
+            setup_func = this.data[FindCollectionNumber(this.data, collection_version)].setupfunc;
+        }
+
+        switch (setup_func) {
+            case "movecfgs":
+                ElectronLog.log("Executing movecfgs install operation")
+                let filesToMove = ["autoexec.cfg", "scout.cfg", "soldier.cfg", "pyro.cfg", "demoman.cfg", "heavyweapons.cfg",
+                "engineer.cfg", "medic.cfg", "sniper.cfg", "spy.cfg"]
+                //Move cfgs
+                let cfgpath = path.join(Main.config.tf2_directory, "tf", "cfg");
+                let usercfgpath = path.join(cfgpath, "user");
+                ElectronLog.log("cfg path is: " + cfgpath);
+                ElectronLog.log("user cfg path is: \"" + usercfgpath + "\"");
+                if (!(fs.existsSync(usercfgpath))) {
+                    //Create usercfgpath
+                    ElectronLog.log("user cfg path does not exist, creating...");
+                    fs.mkdirSync(usercfgpath, {recursive: true});
+                }
+                //Actually move them
+                fs.readdir(cfgpath, (err, files) => {
+                    if (err) {
+                        ElectronLog.error("there was an error on mod_manager.ts, line 447 when trying to read the cfg directory to move cfg files. Trying to read directory: " + cfgpath + " , the error was:" + err)
```

Comment:
I think for these errors, we want to display to the user that something went wrong.
You can use Utilities.ErrorDialog() for this and the isDev package to change the message to a more dev-centric one for development versions.